component/loki/process/stages/extensions.go

```diff
@@ -24,7 +27,37 @@ type DockerConfig struct{}
 
 // CRIConfig is an empty struct that is used to enable a pre-defined pipeline
 // for decoding entries that are using the CRI logging format.
-type CRIConfig struct{}
+type CRIConfig struct {
+	MaxPartialLines            int              `river:"max_partial_lines,attr,optional"`
+	MaxPartialLineSize         flagext.ByteSize `river:"max_partial_line_size,attr,optional"`
```

Comment:
Using flagext.ByteSize here doesn't work because it doesn't implement encoding.TextMarshaler/encoding.TextUnmarshaler and a uint64 can't be directly assigned to it. 

While we could use the underlying ByteSize type, I think we should swap this out for the underlying uint64 since the Loki docs don't mention that this _could_ be a string. 

Since I'm working on helping the tests pass, I'll push this code for you and you can decide if it's OK tomorrow.

Reply:
> Since I'm working on helping the tests pass, I'll push this code for you and you can decide if it's OK tomorrow.

Also, sorry for jumping in. I saw the PR was approved but a test was failing and thought it'd be a quick fix so we can get this merged, but it turned out to be a bit more than I had anticipated.

Reply:
Thanks for looking at this, and no need to apologise :)
I was looking at the same thing yesterday, reached the same conclusion, and had prepared a very similar change which wasn't pushed. Yours looks a bit nicer, so I'm happy to go with your suggestion.